libcudacxx/include/cuda/std/__cuda/semaphore.h

```diff
@@ -23,6 +24,19 @@
 
 _LIBCUDACXX_BEGIN_NAMESPACE_CUDA
 
+void simple_backoff()
+{
+    int wait_iterations = 1;
```

Comment:
This backoff strategy is actually needed, I've found, since `wait()` is not suitable for purposes in `acquire_slow_timed`. No one will notify the thread that is `wait`ing, and as such it will hang. I've discovered this through the test.

Reply:
Is there a way to get `nanosleep` to work on CPU code though? (now that i think about it, I'm surprised this code even compiles given i haven't annotated it with `__device__` )